### DISCUSSION TITLE
New Release of UVdesk Opensource with PHP 8 | Version 1.0.18

### DISCUSSION THREAD

**Discussion by papnoisanjeev** [uvdesk/community-skeleton#493] [Announcements]
**When creating a New project:-**
========================================
**Requirement:**  PHP version  **7.3 or greater** (Will support PHP 8 as well now).

**If you are creating a New Project using command then follow below steps:-**

* **composer clear-cache**
* **composer create-project uvdesk/community-skeleton helpdesk-project**

**When Updating a Existing Project:-**
======================================================
For **PHP 8** we have done some changes in core files as well as updated symfony version.

So you need to perform below steps:

Remove **composer.lock** and  **vendor** folder from your project and replace your **composer.json** file from [here](https://github.com/uvdesk/community-skeleton/blob/master/composer.json).

**Locate path in your project:**

* **src/Controller/ConfigureHelpdesk.php**  open **ConfigureHelpdesk.php** and replace your **ConfigureHelpdesk.php** file from [here](https://github.com/uvdesk/community-skeleton/blob/master/src/Controller/ConfigureHelpdesk.php).
 

**Locate path in your project:**

* **src/DataFixtures/AppFixtures.php** open **AppFixtures.php** and replace your **AppFixtures.php** file from [here](https://github.com/uvdesk/community-skeleton/blob/master/src/DataFixtures/AppFixtures.php).
 

**Locate path in your project:**

* **templates/errors/error.html.twig open error.html.twig** and replace your **error.html.twig** file from [here](https://github.com/uvdesk/community-skeleton/blob/master/templates/errors/error.html.twig).

**Copy all translation files from [here](https://github.com/uvdesk/community-skeleton/tree/master/translations
) into translation folder of your project.**

Take a backup of below files and folder before running below given command:
-------------------------------------------------------------------------------------------------------------
**swiftmailer.yaml, uvdesk_mailbox.yaml, uvdesk.yaml, translation.yaml, services.yaml** (You will find these files under **config->packages** of your project) **public** folder of project.

After you upgraded you will need to replace all your above files along with **public** folder as well with updated project.
 

Finally Run below commands:
 
**composer clear-cache
composer require doctrine/doctrine-cache-bundle
php bin/console d:s:u --force
php bin/console assets:install
php bin/console c:c**
 
**PS:**  **v1.0.17** and **v1.0.18** are same just there was a issue while creating project using command in **v1.0.17** so we have resolved it and again made a new release (v1.0.18).
 
If  Still someone facing any issue we would :heart:  to help you :) 

Thank you !!

**Comment by PeopleInside** [comment]
Thanks, for me the new version still generate a fatal error if the error page 404 is showed.
After I click on a link in the 404 page my server give an error and helpdesk turn down.

I do this:
After following all updating instruction given by you I try to load an inexistent URL like https://helpdesk.peopleinside.it/it/dtrtrt
than I click on one of the links in the new 404 page and server get down with a fatal error.

![uvdesk new error](https://user-images.githubusercontent.com/5006150/139229476-029798b5-ecdb-4ac4-a6c4-ccdbce020ba7.gif)

**Comment by papnoisanjeev** [comment]
@PeopleInside 

We have updated zip on offcial site please try with as well once.
https://www.uvdesk.com/en/opensource/

**Comment by PeopleInside** [comment]
@papnoisanjeev with the FTP version I get this

![Immagine 2021-10-29 143244](https://user-images.githubusercontent.com/5006150/139435022-65ae3f63-bb55-45a7-bdef-e9da00270ff0.jpg)

**Comment by papnoisanjeev** [comment]
@PeopleInside 

open **src** folder in your project and you will see two files with same name **kernel.php**
Remove one of them.

Finally clear your project cache and make a try again.

**Comment by PeopleInside** [comment]
@papnoisanjeev done, deleted one of the two kernel.php file, cleared cache but issue on prod still persist.

**Comment by PeopleInside** [comment]
https://github.com/uvdesk/community-skeleton/issues/495#issuecomment-960294432

**Comment by papnoisanjeev** [comment]
@PeopleInside 

ok then wait we will test it on our server with latest release if issue persist with project itself or not.
On localhost it is working fine.

**Comment by PeopleInside** [comment]
@papnoisanjeev I cannot do nothing else than wait.

I am able to reproduce the issue on upgrading but also on new install from Softaculous.
I'm running Webuzo panel developed by the Webuzo team at Softacolous and they has been stressed by me about this issue: they tested and told they are able to reproduce the issue also on CPanel so the issue is from UVdesk on what they said.

Also I have no issue with UVdesk 1.0.16, no issue with Wordpress and any other PHP application that I run.

**Comment by PeopleInside** [comment]
Isn't curios I get this issue only on prod mode?
As always dev mode seems have no issues... why I'm asking...
I cannot run the help desk on dev mode for security reason.

**Comment by papnoisanjeev** [comment]
@PeopleInside 

>Isn't curios I get this issue only on prod mode?

We will surely fix it once we reproduce this on our server with prod mode.

**Comment by papnoisanjeev** [comment]
@PeopleInside 

We have given you the project link for our live server (with php version 7.4.13) with latest release of project with **prod** mode.
Hope you are not facing any issue during testing.

If any please report back to us we will surely fix it.

**Comment by PeopleInside** [comment]
@papnoisanjeev thanks for the help.
The link provided by you works.

I'm really shocked by how my experience with Webuzo support is going,
now I will continue to asking help to the Webuzo team.

Sorry...
